src/dotnet/APIView/APIViewWeb/Repositories/DevopsArtifactRepository.cs

```diff
@@ -70,7 +70,14 @@ private async Task<HttpResponseMessage> GetFromDevopsAsync(string request)
             int count = 0;
             while ((downloadResp.StatusCode == HttpStatusCode.TooManyRequests || downloadResp.StatusCode == HttpStatusCode.BadRequest) && count < 5)
             {
-                var retryAfter = (downloadResp.Headers.RetryAfter is null) ? "10" : downloadResp.Headers.RetryAfter.ToString();
+                _telemetryClient.TrackTrace($"X-RateLimit-Limit: {downloadResp.Headers.GetValues("X-RateLimit-Limit")}, X-RateLimit-Resource: {downloadResp.Headers.GetValues("X-RateLimit-Resource")}");
+
+                string retryAfter = "300"; // DevOps uses sliding 5 minuites window. At which point we are sure that Rate limit will be reset.
```

Comment:
So we believe the root cause is that the response does not include a `retry-after` header, and our default timeout of 10 seconds needs to be increased to 5 minutes?

Reply:
That seems to be the case, because am able to trigger the request successfully after all retires have failed.